lib/util.ts

```diff
@@ -48,19 +48,18 @@ export function runFilenameOrFn_(
       filenameOrFn = require(resolve(configDir, filenameOrFn));
     }
     if (typeof filenameOrFn === 'function') {
-      let results =
-          when(filenameOrFn.apply(null, args), null, (err: string | Error) => {
-            if (typeof err === 'string') {
-              err = new Error(err);
-            } else {
-              err = err as Error;
-              if (err.stack) {
-                err.stack = new Error().stack;
-              }
-            }
-            err.stack = exports.filterStackTrace(err.stack);
-            throw err;
-          });
+      let results = when(filenameOrFn.apply(null, args), null, (err) => {
```

Comment:
Looked into this and DefinitelyTyped `q.when` has defined a callback `onRejected?: (reason: any)`.